marketplaceItems/avatarCustomization101/appResources/appData/avatarCustomization101.js

```diff
@@ -0,0 +1,937 @@
+            // Update the ui with new properties
+            setMaterialPropertiesToDefaults();
+            updateMaterialDynamicDataUI(newMaterialData, isPBR);
+        } 
+    }
+
+    function prepImageBackendToUI(file) {
+        return file.replace(PATH_TO_IMAGES, "");
+    }
+
+    function setMaterialPropertiesToDefaults () {
+
+        // shadeless
+        setDefaults(dynamicData[STRING_MATERIAL].shadeless, defaultMaterialProperties.shadeless);
+        // pbr
+        setDefaults(dynamicData[STRING_MATERIAL].pbr, defaultMaterialProperties.pbr);
+
+        function setDefaults(dynamicObject, defaultObject) {
+            for (var key in defaultObject) {
+                var defaultValue = defaultObject[key].value;
+                var defaultMap = defaultObject[key].map;
+    
+                dynamicObject[key].value = defaultValue;
+                dynamicObject[key].map = defaultMap;
+            }
+        }
+
+    }
+
+    // presets
+    function applyNamedMaterial(materialName) {
+
+        dynamicData[STRING_MATERIAL].selectedMaterial = materialName;
+
+        switch (materialName){
+            case STRING_DEFAULT_MAT:
+
+                setMaterialPropertiesToDefaults();
+                dynamicData[STRING_MATERIAL].selectedTypeIndex = 0; // "Select one"
+
+                if (materialID) {
+                    Entities.deleteEntity(materialID);
+                    materialID = null;
+                }
+
+                break;
+            case STRING_GLASS_MAT:
+                // updateMaterial materialName, isNamed, isPBR
+                updateMaterial(MATERIAL_GLASS, true, true);
+                break;
+            case STRING_CHAIN_MAT:
+                updateMaterial(MATERIAL_CHAINMAIL, true, true);
+                break;
+            case STRING_DISCO_MAT:
+                updateMaterial(MATERIAL_DISCO, true, true);
+                break;
+            case STRING_RED_MAT:
+                updateMaterial(MATERIAL_RED, true, false);
+                break;
+            case STRING_TEXTURE_MAT:
+                updateMaterial(MATERIAL_TEXTURE, true, false);
+                break;
+        }
+    }
+
+    // #endregion MATERIAL
+
+    // #region BLENDSHAPES
+
+    var TRANSITION_TIME_SECONDS = 0.25;
+    var STRING_AWE = "awe";
+    var STRING_ANGRY = "angry";
+    var STRING_DEFAULT = "default";
+    var STRING_LAUGH = "laugh";
+    var BLENDSHAPES_DEFAULT = BLENDSHAPE_DATA.defaults;
+    var BLENDSHAPES_AWE = BLENDSHAPE_DATA.awe;
+    var BLENDSHAPES_LAUGH = BLENDSHAPE_DATA.laugh;
+    var BLENDSHAPES_ANGRY = BLENDSHAPE_DATA.angry;
+
+    var lastEmotionUsed = BLENDSHAPES_DEFAULT; // values associated with the last emotion lerping
+    var isChangingEmotion = false; // transitioning
+    var changingEmotionPercentage = 0.0; // with transitioning
+    var isBlendshapeIntervalEnabled = false;
+
+    // used to mix between blendshape expressions
+    function mixBlendshapesInterval (deltaTime) {
```

Comment:
please check your spacing between the function name and the open parens on this function and the rest in this file - i notice many inconsistencies in here.